server/server.js

```diff
@@ -1,15 +1,67 @@
 const express = require("express");
 const app = express();
+const videosData = require("./exampleresponse.json");
+app.use(express());
+app.use(express.json());
+app.use(express.urlencoded({ extended: false }));
 const port = process.env.PORT || 5000;
 
 app.listen(port, () => console.log(`Listening on port ${port}`));
 
 // Store and retrieve your videos from here
 // If you want, you can copy "exampleresponse.json" into here to have some data to work with
-let videos = [];
+let videos = [videosData];
 
 // GET "/"
+// app.get("/", (req, res) => {
+//   // Delete this line after you've confirmed your server is running
+//   res.send({ express: "Your Backend Service is Running" });
+// });
+
 app.get("/", (req, res) => {
-  // Delete this line after you've confirmed your server is running
-  res.send({ express: "Your Backend Service is Running" });
+  res.send(videosData);
+});
+
+app.get("/:id", (req, res) => {
+  const videoId = parseInt(req.params.id);
+
+  res.send(videosData[videoId]);
+});
+
+app.post("/videos", (req, res) => {
+  const addNewVideo = req.body;
+
+  addNewVideo.id = Math.floor(Math.random() * 100000);
+  if (!addNewVideo.title || !addNewVideo.url) {
+    res.status(400).send({ msg: "Please add a Title & URL from Youtube !" });
+  } else {
+    videos.push(addNewVideo);
+    res.status(200).send({ msg: `Video:${addNewVideo.title} has been added.` });
+  }
+});
+
+app.get("/videos/:id", (req, res) => {
+  const videoId = parseInt(req.params.id);
+
+  const filterVideo = videosData.find((video) => video.id === videoId);
+
+  if (filterVideo.length === 0) {
```

Comment:
looks good but have you tried testing this, filterVideo may not be an array but this may work but not in the way you intend:

e.g 
```javascript
const letter = ["a", "b", "c"].find((char)=> char === "c")  
```
will have the variable `letter` containing the value "c" not an array, you may want to use `filter` instead of `find`